client/testnet.go

```diff
@@ -0,0 +1,436 @@
+	cmd.Flags().String(flagKeyAlgo, string(crypto.EthSecp256k1), "Key signing algorithm to generate keys for")
+	return cmd
+}
+
+// InitTestnet initializes the testnet configuration
+func InitTestnet(
+	cmd *cobra.Command,
+	config *tmconfig.Config,
+	cdc *codec.Codec,
+	mbm module.BasicManager,
+	genAccIterator authtypes.GenesisAccountIterator,
+	outputDir,
+	chainID,
+	minGasPrices,
+	nodeDirPrefix,
+	nodeDaemonHome,
+	nodeCLIHome,
+	startingIPAddress,
+	keyringBackend,
+	algo string,
+	numValidators int,
+) error {
+
+	if chainID == "" {
+		chainID = fmt.Sprintf("%d", tmrand.Int63())
+	}
+
+	nodeIDs := make([]string, numValidators)
+	valPubKeys := make([]tmcrypto.PubKey, numValidators)
+
+	simappConfig := srvconfig.DefaultConfig()
+	simappConfig.MinGasPrices = minGasPrices
+
+	var (
+		genAccounts []authexported.GenesisAccount
+		genFiles    []string
+	)
+
+	inBuf := bufio.NewReader(cmd.InOrStdin())
+	// generate private keys, node IDs, and initial transactions
+	for i := 0; i < numValidators; i++ {
+		nodeDirName := fmt.Sprintf("%s%d", nodeDirPrefix, i)
+		nodeDir := filepath.Join(outputDir, nodeDirName, nodeDaemonHome)
+		clientDir := filepath.Join(outputDir, nodeDirName, nodeCLIHome)
+		gentxsDir := filepath.Join(outputDir, "gentxs")
+
+		config.SetRoot(nodeDir)
+		config.RPC.ListenAddress = "tcp://0.0.0.0:26657"
+
+		if err := os.MkdirAll(filepath.Join(nodeDir, "config"), nodeDirPerm); err != nil {
+			_ = os.RemoveAll(outputDir)
+			return err
+		}
+
+		if err := os.MkdirAll(clientDir, nodeDirPerm); err != nil {
+			_ = os.RemoveAll(outputDir)
+			return err
+		}
+
+		config.Moniker = nodeDirName
+
+		ip, err := getIP(i, startingIPAddress)
+		if err != nil {
+			_ = os.RemoveAll(outputDir)
+			return err
+		}
+
+		nodeIDs[i], valPubKeys[i], err = genutil.InitializeNodeValidatorFiles(config)
+		if err != nil {
+			_ = os.RemoveAll(outputDir)
+			return err
+		}
+
+		memo := fmt.Sprintf("%s@%s:26656", nodeIDs[i], ip)
+		genFiles = append(genFiles, config.GenesisFile())
+
+		kb, err := keys.NewKeyring(
+			sdk.KeyringServiceName(),
+			keyringBackend,
+			clientDir,
+			inBuf,
+			crypto.EthSecp256k1Options()...,
+		)
+		if err != nil {
+			return err
+		}
+
+		cmd.Printf(
+			"Password for account '%s' :\n", nodeDirName,
+		)
+
+		keyPass := clientkeys.DefaultKeyPass
+		addr, secret, err := GenerateSaveCoinKey(kb, nodeDirName, keyPass, true, keys.SigningAlgo(algo))
+		if err != nil {
+			_ = os.RemoveAll(outputDir)
+			return err
+		}
+
+		info := map[string]string{"secret": secret}
+
+		cliPrint, err := json.Marshal(info)
+		if err != nil {
+			return err
+		}
+
+		// save private key seed words
+		if err := writeFile(fmt.Sprintf("%v.json", "key_seed"), clientDir, cliPrint); err != nil {
+			return err
+		}
+
+		accTokens := sdk.TokensFromConsensusPower(1000)
+		accStakingTokens := sdk.TokensFromConsensusPower(5000)
+		coins := sdk.NewCoins(
+			sdk.NewCoin(sdk.DefaultBondDenom, accTokens),
```

Comment:
the staking token needs to be changed to `"ara"`. Can you create a constant for that?

Reply:
sorry this applies to the staking token below